src/app/saved-jobs/page.tsx

```diff
@@ -1,10 +1,69 @@
+'use client';
 import Link from 'next/link';
+import { ScrollArea } from '../../components/ui/scroll-area';
+import SavedJob from './components/SavedJob';
+import ButtonWithBadge from './components/ButtonWithBadge';
+import { useState } from 'react';
+
+import type { Job } from './placeHolderData';
+import { jobsArr, jobsAmounts } from './placeHolderData';
 
 export default function SavedJobsPage() {
+  const [currentTab, setCurrentTab] = useState<string>('Saved');
+  const filteredArray: Array<Job> = jobsArr.filter(
+    (job) => job.savedLocation == currentTab,
+  );
+
   return (
-    <main>
-      <div className="container flex flex-col items-center justify-center gap-12 px-4 py-16">
-        <h5>Saved jobs page!</h5>
+    <main className='flex justify-center h-full'>
+      <div className="flex w-full sm:w-3/4 md:1/2 h-full flex-col items-center justify-center gap-3 px-4 py-16">
+        <div className="container flex flex-col gap-3">
+          <h2 className="self-start text-4xl sm:text-5xl font-extrabold">Bookmarked Jobs</h2>
+          <div className="flex gap-2 md:gap-3 border border-b-4 border-transparent border-b-inherit p-1">
+            <ButtonWithBadge
+              isTextBold={currentTab == 'Saved'}
+              badgeText={jobsAmounts.savedJobs}
+              onClick={() => setCurrentTab('Saved')}
+            >
+              Saved
+            </ButtonWithBadge>
+            <ButtonWithBadge
+              isTextBold={currentTab == 'Applied'}
+              badgeText={jobsAmounts.appliedJobs}
+              onClick={() => setCurrentTab('Applied')}
+            >
+              Applied
+            </ButtonWithBadge>
+            <ButtonWithBadge
+              isTextBold={currentTab == 'Interviews'}
+              badgeText={jobsAmounts.interviewsJobs}
+              onClick={() => setCurrentTab('Interviews')}
+            >
+              Interviews
+            </ButtonWithBadge>
+            <ButtonWithBadge
+              isTextBold={currentTab == 'Archived'}
+              badgeText={jobsAmounts.archivedJobs}
+              onClick={() => setCurrentTab('Archived')}
+            >
+              Archived
+            </ButtonWithBadge>
+          </div>
+        </div>
+       {filteredArray.length > 0 ?  
+       <ScrollArea className="container h-72 md:h-96 w-full">
+          {filteredArray.map((job) => (
+            <SavedJob
+              key={job.jobTitle}
+              jobTitle={job.jobTitle}
+              company={job.company}
+              location={job.location}
+              savedDate={job.savedDate}
+              companyImage={job.companyImage}
+            />
+          ))}
```

Comment:
It's important to keep the main JSX templating as clean as possible and not include any JS logic like ternaries and maps if we can help it. Usually I move stuff like this into it's own function and would call it something like `renderSavedJobs`